### DISCUSSION TITLE
Problema Termostato Tuya Imported

### DISCUSSION THREAD

**Discussion by mriitan** [indomus/forum#1379] [HUB: Home Assistant]
Ciao a tutti,
ho integrato tramite Tuya un termostato che uso regolarmente, che mi viene anche riconosciuto come funzionamento accensione e spegnimento, ma la temperatura è errata, parte dal valore 1 fino al 7, non è espressa in gradi, come posso modificare la cosa?

![image](https://user-images.githubusercontent.com/98173792/150691340-a64cabae-f404-4c69-b14c-60d884d09b16.png)

vedo questi dettagli sul dispositivo:

climate.termostato
Termostato | heat_cool | 
hvac_modes: off, heat_cool 
min_temp: 1 
max_temp: 7 
target_temp_step: 0.5 
current_temperature: null 
temperature: 1 
friendly_name: Termostato 
supported_features: 1
-- | -- | --

dove posso modificare questi parametri se si possono modificare o come devo agire?
grazie

**Comment by snakuzzo** [comment]
Se hai integrato con l'integrazione Tuya, non penso tu possa modificarli

**Reply by mriitan** [reply]
Sono riuscito a sistemare la cosa, nel senso che se vado negli strumenti per sviluppatore e modifico min temp e max temp con i valori minimi e massimo (9-32) che vedo nell'app del termostato questi si settano su home assistant fino alla ripartenza dell'applicazione. Volendo far caricare quei valori ad ogni partenza dell'applicazione come posso fare? devo inserire questi dettagli nel file configuration.yaml?
qualcuno potrebbe indicarmi la sintassi ?
grazie

**Reply by snakuzzo** [reply]
Quella modifica che hai fatto dovrebbe essere solo temporanea e non persistente. Chi comanda è l'integrazione. Secondo me non puoi farlo programmaticamente in maniera corretta

**Reply by mriitan** [reply]
però resta attiva finchè non riparte l'applicazione, sto facendo tante prove di varie cose, per cui avevo pensato che all'apertura potessi far caricare quegli stati al sensore, no?

**Reply by snakuzzo** [reply]
sarebbe una pezza...anziché risolvere solo per il tuo caso specifico, al posto tuo aprirei anche una issue sul progetto github di core...così magari sarà utile anche per altri.
Quanto alla modifica dello stato, bisogna vedere un attimo se c'è un servizio o qualcosa da poter utilizzare programmaticamente

**Reply by italodelmanso1975** [reply]
Ciao, avevo un problema analogo e non riuscivo a risolvere. Poi ho visto che esiste un'integrazione "Tuya Local" e ho provato ad integrare il termostato e funziona perfettamente. Prova, non so se risolvi anche tu, ovviamente è meglio eliminare la vecchia integrazione. Attenzione a non confonderti con la più famosa integrazione "Local Tuya", quella che ho usato io è quella che ti ho indicato all'inizio (funziona bene (anzi per alcuni versi meglio) anche con altri dispositivi Tuya come prese ed altro)

**Reply by mriitan** [reply]
trovo molto complicato integrare questo soluzione, lo farò in ultima analisi, grazie comunque

**Comment by nonsviluppo** [comment]
buona sera a tutti, anche io ho lo stesso problema di mriitan qualcuno è riuscito a risolverla mille grazie

**Reply by Franco2255** [reply]
Stesso problema ho contattato TUYA SMART e questa è stata la risposta
Tuya engineer11-20 03:13:31
hi there
we will explain it in detail
1. The data reported by the device to Tuya Cloud is actually inconsistent with the data displayed on the APP because the calculation logic is completed on the APP panel. What is displayed in HomeAssistant is the data that the device actually reports to Tuya Cloud. You can call the device log API to refer to the calculation logic for conversion and keep the data consistent with the data on the App panel.
2. This is a multiple relationship formulated by the product creator and cannot be modified by others. Therefore, you need to convert according to the result returned by the interface, such as temperature setting: panel display temperature = temperature value sent by the interface * 0.5

**Reply by Franco2255** [reply]
se moltiplichi la temperatura visualizzata in HA per 5 ti da la temperature riportata sul display.

**Comment by nonsviluppo** [comment]
Ciao, riusciresti a fare una piccola guida passo passo in modo che anche chi come me non è molto smanettone riesce a risolvere.Saresti gentilissimo Inviato dal mio Galaxy
-------- Messaggio originale --------Da: Franco2255 ***@***.***> Data: 20/11/23  20:35  (GMT+01:00) A: indomus/forum ***@***.***> Cc: nonsviluppo ***@***.***>, Comment ***@***.***> Oggetto: Re: [indomus/forum] Problema Termostato Tuya Imported (Discussion #1379) 
se moltiplichi la temperatura visualizzata in HA per 5 ti da la temperature riportata sul display.

**Reply by Franco2255** [reply]
scusa ma io non sono in grado di farlo bisognerebbe chiede a Snakuzzo se si può intervenire in qualche modo

**Reply by Federfico87** [reply]
ciao io sono riuscito a farlo utilizzando local tuya e andando a modificare le proprietà nel file climate.py, ma il problema è che la moltiplicazione della temperatura target avviene anche in scrittura sul dispositivo...quindi posso leggere la temperatura reale che leggo sul display moltiplicando x 5, ma non posso settare la temperatura reale perché altrimenti mi si moltiplica x 5...
